### PR TITLE
Duluth 2024

### PR DESCRIPTION
**Pull request by Dylan-Powers**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -15,6 +15,13 @@
 public final class Constants {
     public static final double stickDeadband = 0.1;
 
+    public static final double ELEVATOR_GEAR_RATIO = 45.0;
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/Constants.java]: Gear ratio is not used anywhere. Should probably be removed.

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -34,6 +42,10 @@ public class Shooter extends SubsystemBase {
   public final double reverseShooterVoltage = 3.0;
   public final double stopShooterVoltage = 0.0;
 
+  public final double runShooterControl = 90;
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/subsystems/Shooter.java]: These three are not used, they should be deleted

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -112,9 +153,42 @@ public RobotContainer() {
...
+
+        NamedCommands.registerCommand("Intake", new RunIntake(s_Intake, s_ShooterPivot, s_Shooter, s_Eyes).until(() -> !s_Shooter.getBreakBeamOutput()));
+        NamedCommands.registerCommand("Score", AimThenShoot);
+        NamedCommands.registerCommand("Score Far", AimThenShootFar);
+        NamedCommands.registerCommand("Aim", new AimShoot(s_Eyes, s_ShooterPivot, s_Shooter));
+        NamedCommands.registerCommand("Fire", new InstantCommand(() -> s_Shooter.setLoaderVoltage(s_Shooter.runLoaderVoltage)));
+        
+        autoChooser = AutoBuilder.buildAutoChooser();
+
+        //autoChooser.addOption("Duluth Auto", new PathPlannerAuto("Duluth Auto"));
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -124,62 +198,146 @@ public RobotContainer() {
      * edu.wpi.first.wpilibj2.command.button.JoystickButton}.
      */
     private void configureButtonBindings() {
-        /* Driver Buttons */
+
+        /*Operator Buttons */
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -124,62 +198,146 @@ public RobotContainer() {
...
+                () -> driver.getRawAxis(leftX), 
+                () -> driver.getRawAxis(rightX),
                 () -> driverDpadUp.getAsBoolean(),
-                () -> s_Swerve.getTargetRotation(),
+                () -> s_Eyes.getTargetRotation(),
                 () -> driverLeftTrigger.getAsBoolean(),
                 rotationSpeed,
                 true
-            )//.until(() -> Math.abs(s_Swerve.getGyroYaw().getDegrees() % 360) < s_Swerve.getTargetRotation() + Constants.AUTO_ROTATE_DEADBAND && 
+            ).alongWith(new AimShoot(s_Eyes, s_ShooterPivot, s_Shooter))//.until(() -> Math.abs(s_Swerve.getGyroYaw().getDegrees() % 360) < s_Swerve.getTargetRotation() + Constants.AUTO_ROTATE_DEADBAND &&
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -124,62 +198,146 @@ public RobotContainer() {
...
+        )).onFalse(
+            new ParallelCommandGroup(
+                new InstantCommand(() -> s_Intake.setIntakePivotPosition(s_Intake.intakeSafePosition)),
+                new InstantCommand(() -> s_ShooterPivot.moveShooterPivot(s_ShooterPivot.shooterPivotStowPosition)),
+                new InstantCommand(() -> s_Intake.setIntakeVoltage(s_Intake.stopIntakeVoltage)),
+                new InstantCommand(() -> s_Shooter.setLoaderVoltage(s_Shooter.stopLoaderVoltage))
+            )
+        );
+
+        /*
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -124,62 +198,146 @@ public RobotContainer() {
...
+                    new AmpShooterPivotRetract(s_ShooterPivot),
+                    s_ShooterPivot.ShooterPivotAtPosition(),
+                    new AmpElevatorRetract(s_Elevator)
+                )
+        );
+
+        // shoot amp
+        operatorRightTrigger.onTrue(
+            new ParallelCommandGroup(
+                //new InstantCommand(() -> s_Shooter.setShooterVoltage(3, -3)),
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -124,62 +198,146 @@ public RobotContainer() {
...
+                new InstantCommand(() -> s_Shooter.setLoaderVoltage(-6))
+            )
+        ).onFalse(
+            new ParallelCommandGroup(
+                //new InstantCommand(() -> s_Shooter.setShooterVoltage(0, 0)),
```
Comment (Dylan-Powers) [inline src/main/java/frc/robot/RobotContainer.java]: remove commented code

**Review by Xcloud1145 — APPROVED** [review]
yippe

**Pull request merged**